File: cpp/Medium/102_binary_tree_level_order_traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void readLevel(TreeNode* root, vector<vector<int>>& levels, int i) {
        if(root == NULL) return;
        if(levels.size() <= i) levels.push_back({});
        levels[i].push_back(root->val);
        i++;
        readLevel(root->left, levels, i);
        readLevel(root->right, levels, i);
    }
    vector<vector<int>> levelOrder(TreeNode* root) {
        int i = 0;
        vector<vector<int>> levels;
        if(root == NULL) return levels;
        readLevel(root, levels, i);
        return levels;
    }
};

File: python/medium/102_binary_tree_level_order_traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def traversal(self, root, level, mapa):
        if root == None: return
        level +=1
        mapa[level].append(root.val)
        self.traversal(root.left, level, mapa)
        self.traversal(root.right, level, mapa)

    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        mapa = defaultdict(list)
        level = 0
        answer = []
        self.traversal(root, level, mapa)
        for i in mapa.values():
            answer.append(i)
        return answer
        
